streams/writable-streams/write.js

```diff
@@ -164,6 +164,30 @@ promise_test(t => {
       .then(() => promise_rejects(t, new TypeError(), writer.close(), 'close() should be rejected'));
 }, 'when sink\'s write throws an error, the stream should become errored and the promise should reject');
 
+promise_test(t => {
+  const passedError = new Error('horrible things');
```

Comment:
Since both of these have `error.name === 'Error'` the `promise_rejects` tests here are no good. Let's stick to the `error1` + `error2` pattern.